src/App.tsx

```diff
@@ -14,11 +14,66 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string;
+};
+
+class App extends Component<{}, State> {
+  state = {
+    selectedGoods: 'Jam',
+  };
+
+  selectHandler = (good: string) => {
+    this.setState({ selectedGoods: good });
+  };
+
+  clearSelect = () => {
+    this.setState({ selectedGoods: '' });
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    return (
+      <div className="App">
+        <button
+          type="button"
+          onClick={() => {
+            this.clearSelect();
+          }}
```

Comment:
you dont need to call the function here. just pass it and it will be called automatically on click